bootstrap-demo/kubernetes-dns/src/main/scala/akka/cluster/bootstrap/KubernetesHealthChecks.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.cluster.bootstrap
+
+import akka.actor.ActorSystem
+import akka.cluster.{ Cluster, MemberStatus }
+import akka.event.{ Logging, LoggingAdapter }
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.server.Route
+
+class KubernetesHealthChecks(system: ActorSystem) {
+
+  val log: LoggingAdapter = Logging(system, getClass)
+  val cluster = Cluster(system)
+
+  //#health
+  private val readyStates: Set[MemberStatus] = Set(MemberStatus.Up, MemberStatus.WeaklyUp)
+  private val aliveStates: Set[MemberStatus] =
+    Set(MemberStatus.Joining, MemberStatus.WeaklyUp, MemberStatus.Up, MemberStatus.Leaving, MemberStatus.Exiting)
```

Comment:
Hm, indeed we don't want K8s to kill the container while it is leaving/exiting, OTOH we want K8s to start removing the node from the loadbalancer, since otherwise there could be a window in which it still might get traffic while it has been moved to `Down`, right?

When a node is no longer 'alive', shouldn't K8s keep in account a grace period, and even when it decides to kill the node, keep a healthy delay between `SIGTERM` and `SIGKILL`?

Reply:
I believe that is what readiness is for, k8s keeps hitting the readiness probe which will only succeeds in Up and WeeklyUp. So we can be alive but not ready when shutting down

Reply:
Ah yes I incorrectly assumed 'ready' would stop polling as soon as the pod has become ready. I agree this is :+1: then - we'd only need to make sure nodes aren't downed faster than the `readinessProbe` `periodSeconds` plus some buffer.